applications/StructuralMechanicsApplication/custom_advanced_constitutive/multi_linear_isotropic_plane_stress_2d.cpp

```diff
@@ -0,0 +1,172 @@
+    return p_clone;
+}
+
+//*******************************DESTRUCTOR*******************************************
+//************************************************************************************
+
+MultiLinearIsotropicPlaneStress2D::~MultiLinearIsotropicPlaneStress2D()
+{
+}
+
+
+//************************************************************************************
+//************************************************************************************
+
+void MultiLinearIsotropicPlaneStress2D::CalculateElasticMatrix(Matrix& C, ConstitutiveLaw::Parameters& rValues)
+{
+    this->CheckClearElasticMatrix(C);
+
+    const Properties& r_material_properties = rValues.GetMaterialProperties();
+    const double nu = r_material_properties[POISSON_RATIO];
+
+    Vector current_strain = ZeroVector(this->VoigtSize);
+    rValues.GetStrainVector(current_strain);
+
+    const double equivalent_strain = std::sqrt(
+        (1.0-nu+std::pow(nu,2.0))  * std::pow(current_strain[0]+current_strain[1],2.0) -
+        3.0 * std::pow(1.0-nu,2.0) * (current_strain[0]*current_strain[1]-std::pow(current_strain[2]/2.0,2.0))
+    ) / (1.0 - std::pow(nu,2.0));
+
+    const Vector moduli_list(rValues.GetMaterialProperties()[MULTI_LINEAR_ELASTICITY_MODULI]);
+    double equivalent_tangent_modulus(0.0);
+
+    if (equivalent_strain>0.0){
+
+        const Vector strain_list(rValues.GetMaterialProperties()[MULTI_LINEAR_ELASTICITY_STRAINS]);
+        const SizeType len_strain_list(strain_list.size());
+
+        SizeType counter(0);
+        for (SizeType i=0;i<len_strain_list;i++){
+            if (equivalent_strain>=strain_list[len_strain_list-(i+1)]){
+                counter = len_strain_list-(i+1);
+                break;
+            }
+        }
+
+        SizeType start_iteration(0);
+        while (start_iteration<counter){
+            equivalent_tangent_modulus += moduli_list[start_iteration]*(strain_list[start_iteration+1]-strain_list[start_iteration]);
+            start_iteration++;
+        }
+        equivalent_tangent_modulus += moduli_list[counter]*(equivalent_strain-strain_list[counter]);
+        equivalent_tangent_modulus /= equivalent_strain;
+
+    }
+    else equivalent_tangent_modulus = moduli_list[0];
+
+
+    const double c1 = equivalent_tangent_modulus / (1.00 - nu * nu);
+    const double c2 = c1 * nu;
+    const double c3 = 0.5*equivalent_tangent_modulus / (1 + nu);
+
+    C(0, 0) = c1;
+    C(0, 1) = c2;
+    C(1, 0) = c2;
+    C(1, 1) = c1;
+    C(2, 2) = c3;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void MultiLinearIsotropicPlaneStress2D::CalculatePK2Stress(
+    const Vector& rStrainVector,
+    Vector& rStressVector,
+    ConstitutiveLaw::Parameters& rValues
+)
+{
+    Matrix c_mat = ZeroMatrix(this->VoigtSize);
```

Comment:
IDEM with BoundedMAtrix?

Reply:
same here, can't pass a BoundedMatrix to CalculateElasticMatrix